app/models/vm_migrate_task.rb

```diff
@@ -57,7 +62,7 @@ def do_request
                   :relocate
                 elsif respool && host.nil?
                   :relocate
-                else
+                elsif host
                   :migrate
                 end
```

Comment:
Since you've removed the catchall `else`, is there a new catchall?

Reply:
There is no catchall case here. Only do certain thing when the condition meets.